pkgs/games/openra/build-engine.nix

```
@@ -0,0 +1,69 @@
+{ lib, buildDotnetModule, dotnetCorePackages
+, SDL2, freetype, openal, lua51Packages
+}:
+engine:
+
+buildDotnetModule rec {
+  pname = "openra";
+  version = "${engine.name}-${engine.version}";
+
+  src = engine.src;
+  nugetDeps = engine.deps;
+
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
+  dotnet-runtime = dotnetCorePackages.runtime_6_0;
+
+  useAppHost = false;
+
+  dotnetFlags = [ "-p:Version=0.0.0.0" ]; # otherwise dotnet build complains, version is saved in VERSION file anyway
+
+  dotnetBuildFlags = [ "-p:TargetPlaform=unix-generic" ];
+  dotnetInstallFlags = [
+    "-p:TargetPlaform=unix-generic"
+    "-p:CopyGenericLauncher=True"
+    "-p:CopyCncDll=True"
+    "-p:CopyD2kDll=True"
+    "-p:UseAppHost=False"
+  ];
+
+  dontDotnetFixup = true;
+
+  preBuild = ''
+    make VERSION=${version} version
+  '';
+
+  postInstall = ''
+    # Create the file so that the install_data script will not attempt to download it.
+    # TODO: fetch the file and include it
+    touch './IP2LOCATION-LITE-DB1.IPV6.BIN.ZIP'
+
+    # Install all the asset data
+    sh -c '. ./packaging/functions.sh; install_data . "$out/lib/${pname}" cnc d2k ra'
```

Why do we wrap this in sh?

---

That's the way the functions in `functions.sh` are called upstream, and I haven't found a better way to do it

---

They are just called like this because they are wrapped in the makefile.

So we could do the following:
```suggestion
    . ./packaging/functions.sh
    install_data . "$out/lib/${pname}" cnc d2k ra
```

or wrap it in a subshell to not pollute our environment. 
```suggestion
    (
      . ./packaging/functions.sh
      install_data . "$out/lib/${pname}" cnc d2k ra
    )
```